src/sql/workbench/services/profiler/browser/profilerService.ts

```diff
@@ -317,6 +318,18 @@ export class ProfilerService implements IProfilerService {
 		if (fileURIs?.length === 1) {
 			const fileURI = fileURIs[0];
 
+			try {
+				let fileSize = await (await fs.stat(fileURI.fsPath)).size;
+				console.log(fileSize);
+				const fileLimitSize = 1073741824; 	// 1GB
+				if (fileSize > fileLimitSize) {
+					this._notificationService.error(nls.localize('FileTooLarge', "File is too large to open and it may hang ADS. "));
```

Comment:
Don't use ADS in custom-facing messages. Use the full product name.

Please also review this string with other people, it's kind of clunky right now. Something like "The file "<filename>" is too large to be opened by Azure Data Studio."

Having some sort of suggestion for what they can do to fix it wouldn't hurt either...

Reply:
Updating it to "The file is too large to open in profiler".
As of now, there is no other suggestion to fix it, apart from opening in SSMS.

Reply:
Can you at least mention what the max file size is in the error message?

Reply:
Updated it to: "The file is too large to open in profiler. The profiler can open files that are less than 1GB."

Additionally added another notification for files more than 100MB and less than 1GB: "Loading the file might take a moment, because the file is large."

Reply:
Why not have it displaying a toast notification with the progress indicator while it's loading? That's a pattern we use across ADS (for example what Alan just changed to display for scripting operations) and is a much clearer indicator of work being done IMO

Reply:
I can look into it as a separate workitem to add a loading indicator. I am not sure how much of it can be done, because currently STS returns the whole file in one-go, and not in chunks. So, ADS doesn't know until STS has returned something. But yes, this can be implemented on the STS side to show some loading progress.